app/models/invenio_rdm_record_converter.rb

```diff
@@ -288,15 +288,11 @@ def subjects_for_field(terms, field)
       term = term.strip
 
       if term.blank?
-        return nil
-      end
-
-      pid = @@header_lookup.pid_lookup_by_field(term, field)
-
-      if pid.present?
-        {id: pid}
-      elsif field == :subject_name || field == :tag
+        nil
+      elsif field == :tag
         {subject: term.force_encoding("UTF-8")}
+      elsif pid = @@header_lookup.pid_lookup_by_field(term, field)
+        {id: pid}
```

Comment:
So we don't let unfound terms through now? Works for me, let's pay attention if librarians say anything after new export.

Also, I think this is actually a good exception to typically not placing `=` in conditionals: an intensive operation included in larger case branching that would have broken it up otherwise or required expensive recomputation :heavy_check_mark: .